x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/use_columns.tsx

```diff
@@ -195,17 +195,53 @@ const useActionsColumn = (): EuiTableActionsColumnType<Rule> => {
 };
 
 export const useRulesColumns = ({ hasPermissions }: ColumnsProps): TableColumn[] => {
+  const { triggersActionsUi } = useKibana().services;
   const actionsColumn = useActionsColumn();
   const enabledColumn = useEnabledColumn({ hasPermissions });
   const ruleNameColumn = useRuleNameColumn();
-  const { isInMemorySorting } = useRulesTableContext().state;
+  const {
+    state: { isInMemorySorting, loadingRuleIds, isLoading },
+    actions: { reFetchRules },
+  } = useRulesTableContext();
   const [showRelatedIntegrations] = useUiSetting$<boolean>(SHOW_RELATED_INTEGRATIONS_SETTING);
 
   return useMemo(
     () => [
       ruleNameColumn,
       ...(showRelatedIntegrations ? [INTEGRATIONS_COLUMN] : []),
       TAGS_COLUMN,
+      {
+        id: 'ruleSnoozeNotify',
+        name: (
+          <EuiToolTip
+            data-test-subj="rulesTableCell-notifyTooltip"
+            content={i18n.COLUMN_NOTIFY_TOOLTIP}
+          >
+            <span>
+              {i18n.COLUMN_NOTIFY}
+              &nbsp;
+              <EuiIcon size="s" color="subdued" type="questionInCircle" className="eui-alignTop" />
+            </span>
+          </EuiToolTip>
+        ),
+        width: '14%',
+        'data-test-subj': 'rulesTableCell-rulesListNotify',
+        render: (rule: Rule) => {
+          return triggersActionsUi.getRulesListNotifyBadge({
+            rule: {
+              ...rule,
+              muteAll: rule.mute_all ?? false,
+              activeSnoozes: rule.active_snoozes,
+              isSnoozedUntil: rule.is_snoozed_until,
+              snoozeSchedule: rule?.snooze_schedule ?? [],
+              isEditable: hasPermissions,
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            } as any,
+            isLoading: loadingRuleIds.includes(rule.id) || isLoading,
+            onRuleChanged: reFetchRules,
```

Comment:
Why do we need to refetch the current page of rules when only the snooze info can be changed by this component? I mean, it can't change other rule fields and it can't change other rules. Could it be stateful and maintain its own state when calling API endpoints?

Reply:
Yes! we can do that